### ISSUE TITLE
Is BatchNorm2d quantized in tensorflow_quantization?

### ISSUE THREAD

**Issue by ardeal** [NVIDIA/TensorRT#3366]
Hi,

I noticed that BatchNorm2d in pytorch_quantization is not quantized.
Is BatchNorm2d quantized in tensorflow_quantization?
`fine_tune` function in tensorflow_quantization is a black box for user. I don't know whether `BatchNorm2d` is quantized or not.

**Comment by zerollzeng** [comment]
@ttyio ^ ^

**Comment by ttyio** [comment]
Hi @ardeal , no need to quant BN because BN is fused into the previous Conv. Thanks!

**Comment by ardeal** [comment]
@ttyio 
Thank you for your reply!
The further questions are:

1) BN is fused to Conv and quantized with Conv together, right?
2) if I don't want to quantize BN, or if I want to quantize BN individually, how to do it?
3) do you have any complete example about: how to use tensorflow_quantization? how to run int8 inference after QAT?

**Comment by ttyio** [comment]
Hi @ardeal 
1. BN is folded in the Conv weights before Conv quantization.
2. The question is to add Q/DQ pair before arbitral layers?  We need insert `quant_nn.TensorQuantizer` in your model, here is sample code: https://github.com/NVIDIA/TensorRT/blob/release/8.6/tools/pytorch-quantization/examples/torchvision/models/classification/resnet.py#L198, 
some details on the code: https://docs.nvidia.com/deeplearning/tensorrt/pytorch-quantization-toolkit/docs/index.html#further-optimization
3. The doc can be found in https://docs.nvidia.com/deeplearning/tensorrt/tensorflow-quantization-toolkit/docs/docs/basics.html, code in https://github.com/NVIDIA/TensorRT/tree/release/8.6/tools/tensorflow-quantization/examples

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, pls reopen if you still have question, thanks!
